src/Containers/BlockChainTest.cpp

```diff
@@ -47,6 +47,29 @@ class MovableType {
 
 };  // namespace
 
+TEST(BlockChain, CreateAndMove) {
+  BlockChain<int, 1024> source_chain;
+  EXPECT_EQ(source_chain.root(), nullptr);
+  source_chain.emplace_back(1);
+  source_chain.emplace_back(2);
+  EXPECT_NE(source_chain.root(), nullptr);
+
+  BlockChain<int, 1024> target_chain(std::move(source_chain));
+  EXPECT_EQ(target_chain.size(), 2);
+  EXPECT_EQ(target_chain.root()->data()[0], 1);
+  EXPECT_EQ(source_chain.size(), 0);
```

Comment:
Ahhh, I guess, I have to admit that clang-tidy is probably right here. And my suggestion was not that great. :grimacing: Sorry...

Reply:
This doesn't prevent from merging, does it? And while clang-tidy is right to report this, I would ignore the warning because I want to test if the source_chain object has been nicely cleaned up. The standard does not forbid accessing moved-from objects, it's just not defined how they have to behave. And I'm testing they behave as I want them to.

You could argue that the new tests below (decoupling of both objects) isn't meaningful, but it also doesn't hurt.

Reply:
In that case, you can add a comment disabling clang-tidy. Either `// NOLINT(bugprone-use-after-move)` behind the statement of `// NOLINTNEXTLINE(bugprone-use-after-move)` in a separate line before.

Reply:
Done. I've also slightly adjusted the test to cover the case that will actually appear in the wild - modifying the target_chain should not change anything in the source_chain any more.